simulation.hpp

```diff
@@ -237,17 +241,20 @@ class Simulation
             int row_to = moved[2];
             int col_to = moved[3];
             int hosts = moved[4];
-            if (hosts > total_plants(row_from, col_from)) {
-                total_hosts_moved = total_plants(row_from, col_from);
+            if (hosts > total_hosts(row_from, col_from)) {
+                total_hosts_moved = total_hosts(row_from, col_from);
             } else {
                 total_hosts_moved = hosts;
             }
             if (infected(row_from, col_from) > 0 && susceptible(row_from, col_from) > 0) {
-                inf_ratio = double(infected(row_from, col_from)) / double(total_plants(row_from, col_from));
+                inf_ratio = double(infected(row_from, col_from)) / double(total_hosts(row_from, col_from));
                 int infected_mean = total_hosts_moved * inf_ratio;
                 if (infected_mean > 0) {
-                    std::poisson_distribution<int> distribution(infected_mean);
-                    infected_moved = distribution(generator_);
+                	std::poisson_distribution<int> distribution(infected_mean);
+                	infected_moved = movement_stochasticity_;
```

Comment:
Again good catch that this is another area to add a non-stochastic version. Here the infected mean could be moved as the non-stochastic option instead of drawing from a Poisson distribution.

Reply:
Thank you for the corrections. I will make those changes now.